### PR TITLE
Fixes for algorithm formatting and health monitor

### PR DESCRIPTION
**Pull request by hemantavi**
- Algorithm formatting: From the raw response, the algorithm settings
  weren't getting parsed.
- Updating a GS member in case of LB type service updated the health
  monitor to HTTP type health monitor instead of TCP.

### CODE REVIEW

File: gslb/cache/controller_obj_cache.go [hunk 1]
```diff
@@ -421,26 +421,57 @@ func parseDescription(description string) ([]string, error) {
...
-	if gsPool.FallbackAlgorithm != nil {
+	pa := gslbalphav1.PoolAlgorithmSettings{LBAlgorithm: *algorithm}
+	if fallbackAlgorithm != nil {
 		gfa := gslbalphav1.GeoFallback{
-			LBAlgorithm: *gsPool.FallbackAlgorithm,
+			LBAlgorithm: *fallbackAlgorithm,
 		}
-		if gsPool.ConsistentHashMask != nil {
-			hashMask := int(*gsPool.ConsistentHashMask)
+		if consistentHashMask != nil {
```
Comment (parimanur) [inline gslb/cache/controller_obj_cache.go:437]: both the checks ( next else ) can be merged irrespective of fallbackAlgorithm?

File: gslb/cache/controller_obj_cache.go [hunk 2]
```diff
@@ -613,18 +644,14 @@ func GetDetailsFromAviGSLB(gslbSvcMap map[string]interface{}) (uint32, []GSMembe
...
 			gslbutils.Warnf("couldn't parse group: %v", val)
 			continue
 		}
-		gsGroup, ok := val.(models.GslbPool)
-		if !ok {
-			gslbutils.Warnf("couldn't parse group to GslbPool: %v", gsGroup)
-			continue
-		}
-		poolAlgorithmSettings = ParsePoolAlgorithmSettings(gsGroup)
+		poolAlgorithmSettings = ParsePoolAlgorithmSettingsFromPoolRaw(group)
```
Comment (parimanur) [inline gslb/cache/controller_obj_cache.go:654]: check poolAlgorithmSettings is nil / return error to handle the error from function.

File: gslb/cache/controller_obj_cache.go (hunk 1 above)
Comment (hemantavi) [reply gslb/cache/controller_obj_cache.go:437]: I don't think so. `consistentHashMask` has to be checked in different contexts. If its in the context of `fallbackAlgorithm`, then we assign it to the geoFallback struct, or else, we assign it to the normal algorithm settings.

However, I think there's a bug here, where we are not assigning the `geoFallback` to `pa`. Will fix it.

File: gslb/cache/controller_obj_cache.go (hunk 2 above)
Comment (hemantavi) [reply gslb/cache/controller_obj_cache.go:654]: Actually, its taken care of in the checksum calculation which is the only place where this variable is getting used.

Comment (parimanur) [reply gslb/cache/controller_obj_cache.go:655]: Yes. Searched that it was going to the checksum function,  later did not check the usage.

**Review by parimanur — APPROVED** [review]

**Review by chauhanshubham — APPROVED** [review]

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**